src/components/search.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import styles from './search.module.css'
+import { useDebounce } from '../components/hooks/useDebounce'
+import { useRouter } from 'next/router'
+
+export const Search = ({ onClick, content, image, like }) => {
+  const [query, setQuery] = React.useState('')
+  const [results, setResults] = React.useState([])
```

Comment:
As a bonus feature, I decided to run it with the state instead of redux/actions for the diversity of implementation.

Even if you use Redux for an application, it is not necessary to onboarding every state change into the application state and depends on how deep you want to be able to replicate application states, or redux features.